plugins/available/pip.plugin.bash

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+cite about-plugin
+about-plugin 'pip helper functions'
+
+pip-upgrade-all() {
```

Comment:
Could this be renamed to be `user` specific? e.g. `pip-upgrade-all-user`?

Reply:
There are some subtleties here.
I use this command to upgrade outdated pip packages installed by the system package manager to user's home.
If --user is provided, only those installed by `pip install --user ` will be upgraded.
This may be the desired behaviour for some users, but since pip will also install dependencies to user's home, which may in fact upgrade system-installed packages, the option --user does not actually guarantee "only user-installed packages are upgraded".
Maybe we should provide some option for the caller of the command to choose between them?

Reply:
Indeed, the `about` info is quite misleading.
I'm going to come up with a better description.

Reply:
To upgrade all system- and user-installed pip packages to user's home was what this command was originally intended for.

Reply:
> Indeed, the `about` info is quite misleading.
> I'm going to come up with a better description.

You should do that, otherwise looks pretty okay